Refactor PomodoroCommandManager

joelchartier opened the pull request:
- rename PomodoroCommandManager to SlackMessagePostedListenerConfiguration
- Extract logic to a PomodoCommand factory and each command in its own class

Commit:
Refactor PomodoroCommandManager
- rename PomodoroCommandManager to SlackMessagePostedListenerConfiguration
- Extract logic to a PomodoCommand factory and each command in its own class

prichrd commented on src/main/java/com/github/jchartier/slackbot/pomodoro/command/PomodoroCommandFactory.java:
```diff
@@ -0,0 +1,63 @@
+import java.util.regex.Pattern;
+
+@Component
+public class PomodoroCommandFactory {
```
J'aime beaucoup.

pric commented on src/main/java/com/github/jchartier/slackbot/pomodoro/command/impl/HelpCommand.java:
```diff
@@ -0,0 +1,44 @@
+
+public class HelpCommand implements PomodoroCommand {
+
+    // TODO: Centralized this character
```
What does this mean?

prichrd commented on src/main/java/com/github/jchartier/slackbot/pomodoro/command/PomodoroCommand.java:
```diff
@@ -0,0 +1,8 @@
+import com.ullink.slack.simpleslackapi.SlackUser;
+
+public interface PomodoroCommand {
+
```
We should add the help message in every Pomodoro commands. It will help to:
- Build the `help` command
- Give some feedback when the command fails

prichrd commented on src/main/java/com/github/jchartier/slackbot/pomodoro/command/impl/HelpCommand.java:
```diff
@@ -0,0 +1,44 @@
+        stringBuilder.append("\n");
+        stringBuilder.append("Here are the available commands \n");
+        stringBuilder.append(DOT);
+        stringBuilder.append(" `start pomodoro <time_in_minutes>` starts a new pomodoro (between 1 and 999 minutes)} \n");
```
Like I said previously, we could implement a function returning these messages in every commands.

Commit:
Add validation on command + manage command exceptions thrown
WIP: Add start delay validation that throws exception if delay is not between 1 and 999

joelchartier committed:
Merge pull request #15 from joelchartier/validate-start-command

Add validation on command + manage command exceptions thrown
